tracee-ebpf/main.go

```diff
@@ -95,13 +95,19 @@ func main() {
 			if err != nil {
 				return err
 			}
-			err = checkRequiredCapabilities(selfCap)
+			if err = checkRequiredCapabilities(selfCap); err != nil {
+				return err
+			}
+
+			enabled, err := helpers.FtraceEnabled()
 			if err != nil {
 				return err
 			}
+			if !enabled {
+				fmt.Fprintf(os.Stderr, "ftrace_enabled: warning: ftrace is not enabled - kprobes won't be inserted\n")
```

Comment:
To a user, I don't know if having "kprobes won't be inserted" is very useful. Is it possible to list events that won't be traced? Are we sure that ftrace is absolutely required for all kprobes?

Reply:
I think so yeah. @yanivagman ?

Reply:
Ftrace is needed for kprobes as they're built on top of ftrace base architecture (https://www.kernel.org/doc/Documentation/trace/ftrace-design.txt), like an extension to it. I don't think we should expand to much here, but I do think the message could be clearer for an average user that might not know what kprobes and ftrace are.

Reply:
I actually think that if a user is using Tracee then most likely he will have some technical background (and kprobes knowledge as well). I suggest something like `"ftrace_enabled: warning:  ftrace is not enabled - some events may not be caught (kprobes not enabled)"`.

Reply:
I believe no events, at all, will be caught, if ftrace is disabled. 

And I disagree that users of tracee will know what a kprobe (or ftrace) is. 

I do agree that a warning like that could be helpful as long as it is rephrased to something like:

`ftrace_enabled: warning: ftrace is not enabled, kernel events won't be caught, make sure to enable it by executing echo 1 | sudo tee /proc/sys/kernel/ftrace_enabled`

Reply:
You do get events coming from tracepoints. 

Let's go with your message then :)

Reply:
> You do get events coming from tracepoints.

I see what you mean. But our ebpf logic might be broken as there are maps manipulated by different eBPF programs (with different probe methods) in different times, I believe. 

I agree that if we consider tracee purely as an introspection tool, final user might know what we're talking about... but we're trying to shift that mindset as well (thus the explicit message about how to get it enabled).

Cool, thanks.

Reply:
So if this is the case should we just error out if ftrace is disabled? Is there any point in running tracee?

Reply:
> So if this is the case should we just error out if ftrace is disabled? Is there any point in running tracee?

I don't think so - tracepoint will still work, which means that all the syscalls + some other events will still work (> 95% of the events are not kprobes) - so I think we should just warn

Reply:
I believe you're right @grantseltzer. Its better to just fail instead of trying to rely in a tainted execution.

Reply:
Alright then, works for me.